pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -852,6 +856,15 @@ func (gce *GCECloud) createTargetPool(name, serviceName, region string, hosts []
 	for _, host := range hosts {
 		instances = append(instances, makeHostURL(gce.projectID, host.Zone, host.Name))
 	}
+	if len(instances) > maxInstancesPerTargetPool {
```

Comment:
I think that's beyond the scope of this PR. We need something like #23028, which was related to e.g. #24295, to handle that, and would presumably result in the node getting kicked out / rendered unhealthy and removed by the service controller.